pubsub/natspubsub/nats.go

```diff
@@ -160,24 +159,30 @@ func (*topic) ErrorCode(err error) gcerrors.ErrorCode {
 }
 
 type subscription struct {
-	nc   *nats.Conn
-	nsub *nats.Subscription
-	err  error
+	nc      *nats.Conn
+	nsub    *nats.Subscription
+	ackFunc func()
+	err     error
 }
 
 // CreateSubscription returns a *pubsub.Subscription representing a NATS subscription.
 // TODO(dlc) - Options for queue groups?
-func CreateSubscription(nc *nats.Conn, subscriptionName string) *pubsub.Subscription {
-	return pubsub.NewSubscription(createSubscription(nc, subscriptionName), nil)
+func CreateSubscription(nc *nats.Conn, subscriptionName string, ackFunc func()) *pubsub.Subscription {
+	return pubsub.NewSubscription(createSubscription(nc, subscriptionName, ackFunc), nil)
 }
 
-func createSubscription(nc *nats.Conn, subscriptionName string) driver.Subscription {
+func createSubscription(nc *nats.Conn, subscriptionName string, ackFunc func()) driver.Subscription {
 	sub, err := nc.SubscribeSync(subscriptionName)
-	return &subscription{nc, sub, err}
+	return &subscription{nc, sub, ackFunc, err}
 }
 
 // AckFunc implements driver.Subscription.AckFunc.
-func (*subscription) AckFunc() func() { return nil }
+func (s *subscription) AckFunc() func() {
+	if s == nil {
```

Comment:
We generally don't write these receiver nil checks. (Otherwise we'd be writing them everywhere.) Unless there is some good reason, like it's normal for `s` to be nil.

Reply:
The tests crash with out it, here: 

https://github.com/google/go-cloud/blob/master/pubsub/pubsub.go#L537

The conformance tests create a "nonexistent" subscription, which is implemented for NATS with a `nil` `*subscription`; `pubsub.NewSubscription` does get called for it though.